modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1717,11 +1717,21 @@ void GDScriptAnalyzer::reduce_binary_op(GDScriptParser::BinaryOpNode *p_binary_o
 
 	GDScriptParser::DataType left_type;
 	if (p_binary_op->left_operand) {
-		left_type = p_binary_op->left_operand->get_datatype();
+		if (p_binary_op->left_operand->is_constant && p_binary_op->left_operand->reduced) {
```

Comment:
Not sure about the `&& p_binary_op->left_operand->reduced` part. Shouldn't it have been reduced for sure at that point? If so, moving that part of the condition into a `CRASH_COND` or a less harsh `WARN_PRINT` would be better. That would help catching bugs in GDScript.

Reply:
Oh, that is probably me missunderstanding the meaning of **reduced**. Will remove it 👍

Reply:
Maybe I'm on the same boat. I'm just trying to be sure that we who didn't write GDScript understand the role of every piece.